atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientFactory.java

```diff
@@ -208,6 +218,29 @@ private static void login(Client client, CassandraCredentialsConfig config) thro
         client.login(new AuthenticationRequest(credsMap));
     }
 
+    /**
+     * Verifies that the current SSL connection hostname/ip address matches what the certificate has served.
+     * This will check both ip address/hostname, and uses the IP address associated with the socket, rather
+     * that what has been provided. Hostname/ip address are both need to be checked, as historically we've
+     * connected to Cassandra directly using IP addresses, and therefore need to support such cases.
+     */
+    @VisibleForTesting
+    static void verifyEndpoint(CassandraServer cassandraServer, SSLSocket socket, boolean throwOnFailure)
+            throws SafeSSLPeerUnverifiedException {
+        Set<String> validAddresses =
+                Set.of(socket.getInetAddress().getHostAddress(), cassandraServer.cassandraHostName());
```

Comment:
This line is causing production failures when these values are the same.

Caught this in integration tests for our internal authentication service.

```
Caused by: java.lang.IllegalArgumentException: duplicate element: 172.19.0.6
	at java.base/java.util.ImmutableCollections$Set12.<init>(ImmutableCollections.java:791)
	at java.base/java.util.Set.of(Set.java:487)
	at com.palantir.atlasdb.keyvalue.cassandra.CassandraClientFactory.verifyEndpoint(CassandraClientFactory.java:231)
	at com.palantir.atlasdb.keyvalue.cassandra.CassandraClientFactory.getRawClient(CassandraClientFactory.java:183)
	at com.palantir.atlasdb.keyvalue.cassandra.CassandraClientFactory.getRawClientWithTimedCreation(CassandraClientFactory.java:156)
	at com.palantir.atlasdb.keyvalue.cassandra.CassandraClientFactory.getRawClientWithKeyspaceSet(CassandraClientFactory.java:117)
	at com.palantir.atlasdb.keyvalue.cassandra.CassandraClientFactory.create(CassandraClientFactory.java:94)
	... 76 more
```